python/pyiceberg/catalog/__init__.py

```diff
@@ -14,7 +14,233 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-from typing import Dict, Tuple
 
-Identifier = Tuple[str, ...]
-Properties = Dict[str, str]
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from pyiceberg.schema import Schema
+from pyiceberg.table import Table
+from pyiceberg.table.partitioning import PartitionSpec
+
+
+class Catalog(ABC):
+    """Base Catalog for table operations like - create, drop, load, list and others.
+
+    The catalog table APIs accept a table identifier, which is fully classified table name. The identifier can be a string or
+    tuple of strings. If the identifier is a string, it is split into a tuple on '.'. If it is a tuple, it is used as-is.
+
+    The catalog namespace APIs follow a similar convention wherein they also accept a namespace identifier that can be a string
+    or tuple of strings.
+
+    Attributes:
+        name(str): Name of the catalog
+        properties(Dict[str, str]): Catalog properties
+    """
+
+    def __init__(self, name: str, properties: dict[str, str]):
```

Comment:
I don't think that the contents should change when moved. Can you make sure that this is as close as possible to the original?